TruckMuncher/MapViewController.swift

```diff
@@ -148,56 +221,107 @@ class MapViewController: UIViewController, MKMapViewDelegate, CLLocationManagerD
+            }
+            
+            self.truckCarousel.frame = newRect
+            }, completion: nil)
+    }
+    
+    func numberOfItemsInCarousel(carousel: iCarousel!) -> Int
+    {
+        return activeTrucks.count
+    }
+    
+    func carousel(carousel: iCarousel!, viewForItemAtIndex index: Int, var reusingView view: UIView!) -> UIView!
+    {
+        if view == nil {
+            var viewArray = NSBundle.mainBundle().loadNibNamed("TruckDetailView", owner: nil, options: nil)
+            view = viewArray[0] as TruckDetailView
+            view.frame = CGRectMake(0.0, 0.0, UIScreen.mainScreen().bounds.width, UIScreen.mainScreen().bounds.height)
+        }
+        (view as TruckDetailView).updateViewWithTruck(activeTrucks[index])
+
+        return view
+    }
+    
+    func carouselCurrentItemIndexDidChange(carousel: iCarousel!) {
+        var annotations = mapClusterController.annotations.allObjects
+        
+        if (annotations.count > 0) {
+            let curIndex = truckCarousel.currentItemIndex
+            centerMapOverCoordinate(annotations[curIndex].coordinate)
+        }
+    }
+    
+    func carousel(carousel: iCarousel!, didSelectItemAtIndex index: Int) {
+        if !showingMenu {
+            UIView.animateWithDuration(0.5, delay: 0.0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0.5, options: nil, animations: { () -> Void in
+                self.truckCarousel.frame = CGRectMake(0.0, self.mapView.frame.maxY - 130.0, UIScreen.mainScreen().bounds.width, UIScreen.mainScreen().bounds.height)
+            }, completion: { (Bool) -> Void in
+                UIView.animateWithDuration(0.5, animations: { () -> Void in
+                    self.truckCarousel.frame = CGRectMake(0.0, self.mapView.frame.maxY - 100.0, UIScreen.mainScreen().bounds.width, UIScreen.mainScreen().bounds.height)
+                }, completion: nil)
+            })
+        }
+    }
+    
+    // MARK: - UIVieControllerTransitioningDelegate Methods
+    func animationControllerForPresentedController(presented: UIViewController, presentingController presenting: UIViewController, sourceController source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        
+        self.transitionManager.transitionTo = .MODAL;
+        return self.transitionManager;
+    }
+    
+    func animationControllerForDismissedController(dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        self.transitionManager.transitionTo = .INITIAL;
+        return self.transitionManager;
+    }
+    
+    // Mark: - UIGestureRecognizerDelegate Methods
+    func gestureRecognizer(gestureRecognizer: UISwipeGestureRecognizer, shouldRecognizeSimultaneouslyWithGestureRecognizer otherGestureRecognizer: UITapGestureRecognizer) -> Bool{
+            return true;
+    }
+    
+    override func prefersStatusBarHidden() -> Bool {
+        return showingMenu
```

Comment:
This method won't do anything. Dynamically updating the return value of the `prefersStatusBarHidden` method requires you to call `setNeedsStatusBarAppearanceUpdate` every time it changes otherwise that method is called once and never again. Unless something else is implicitly calling the setNeedsStatus... method then this is not needed.